bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHandler.java

```diff
@@ -79,59 +77,106 @@ public ChatGPTHandler(Thing thing, HttpClientFactory httpClientFactory) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        if (command instanceof RefreshType && !"".equals(lastPrompt)) {
-            String response = sendPrompt(channelUID, lastPrompt);
-            processChatResponse(channelUID, response);
-        }
 
         if (command instanceof StringType stringCommand) {
             lastPrompt = stringCommand.toFullString();
-            String response = sendPrompt(channelUID, lastPrompt);
-            processChatResponse(channelUID, response);
+
+            String queryJson = prepareRequestBody(channelUID);
+
+            if (queryJson != null) {
+                String response = sendPrompt(queryJson);
+                processChatResponse(channelUID, response);
+            }
         }
     }
 
     private void processChatResponse(ChannelUID channelUID, @Nullable String response) {
         if (response != null) {
-            ChatResponse chatResponse = gson.fromJson(response, ChatResponse.class);
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            ChatResponse chatResponse;
+            try {
+                chatResponse = objectMapper.readValue(response, ChatResponse.class);
+            } catch (JsonProcessingException e) {
+                logger.error("Failed to parse ChatGPT response: {}", e.getMessage(), e);
+                return;
+            }
+
             if (chatResponse != null) {
-                String msg = chatResponse.getChoices().get(0).getMessage().getContent();
-                updateState(channelUID, new StringType(msg));
+
+                String finishReason = chatResponse.getChoices().get(0).getFinishReason();
+
+                if ("length".equals(finishReason)) {
+                    logger.error("token length exceeded. Increase maximum token to avoid the issue.");
+                    return;
+                }
+
+                @Nullable
+                ChatMessage chatResponseMessage = chatResponse.getChoices().get(0).getChatMessage();
+
+                if (chatResponseMessage == null) {
+                    logger.error("ChatGPT response does not contain a message.");
```

Comment:
What could cause this? Should the user consider this as a bug and file a ticket? If so, error level is ok.

Reply:
It is an unexpected error that occurs only if OpenAI changes something. In that case, we need to know about it as soon as possible.